shared/proto_util.h

```diff
@@ -0,0 +1,142 @@
+#ifndef PROTO_UTIL_H
+#define PROTO_UTIL_H
+
+#include "strings_util.h"
+#include <google/protobuf/descriptor.h>
+
+inline void SetProtoField(google::protobuf::Message* msg,
+                          const google::protobuf::FieldDescriptor* field,
+                          std::string value) {
+  const google::protobuf::Reflection* refl = msg->GetReflection();
+  using CppType = google::protobuf::FieldDescriptor::CppType;
+
+  switch (field->cpp_type()) {
+    case CppType::CPPTYPE_INT32: {
+      refl->SetInt32(msg, field, std::stol(value));
+      break;
+    }
+    case CppType::CPPTYPE_INT64: {
+      refl->SetInt64(msg, field, std::stoll(value));
+      break;
+    }
+    case CppType::CPPTYPE_UINT32: {
+      refl->SetUInt32(msg, field, std::stoul(value));
+      break;
+    }
+    case CppType::CPPTYPE_UINT64: {
+      refl->SetUInt32(msg, field, std::stoull(value));
+      break;
+    }
+    case CppType::CPPTYPE_DOUBLE: {
+      refl->SetDouble(msg, field, std::stod(value));
+      break;
+    }
+    case CppType::CPPTYPE_FLOAT: {
+      refl->SetFloat(msg, field, std::stof(value));
+      break;
+    }
+    case CppType::CPPTYPE_BOOL: {
+      refl->SetBool(msg, field, ParseBool(value));
+      break;
+    }
+    case CppType::CPPTYPE_ENUM: {
+      auto val = field->enum_type()->FindValueByName(value);
+      if (val == NULL)
+        val = field->enum_type()->FindValueByNumber(std::stol(value));
+      if (val != NULL)
+        refl->SetEnum(msg, field, val);
+      break;
+    }
+    case CppType::CPPTYPE_STRING: {
+      refl->SetString(msg, field, value);
+      break;
+    }
+    default:
+      // error
+      break;
+  };
+}
+
+inline void SetProtoField(google::protobuf::Message* msg,
+                          const google::protobuf::FieldDescriptor* field,
+                          double value) {
+  const google::protobuf::Reflection* refl = msg->GetReflection();
+  using CppType = google::protobuf::FieldDescriptor::CppType;
+
+  switch (field->cpp_type()) {
+    case CppType::CPPTYPE_INT32: {
+      refl->SetInt32(msg, field, (int) value);
```

Comment:
some warnings will complain about c-style casts. I would prefer c++ casts here but its just a nit

Reply:
Where did this method come from? It looks odd to me and I don't think I wrote it. Why isn't it just templated?